### PR TITLE
chore: prune dependencies and update to latest versions

### PR DESCRIPTION
**Pull request by safarmer**
Switching to flutter_test from test reduces the transitive dependency graph reducing potential version conflicts

Depends on #41

### CODE REVIEW

File: lib/src/dotenv.dart [hunk 1]
```diff
@@ -49,6 +49,19 @@ class DotEnv {
   /// Clear [env]
   void clean() => _envMap.clear();
 
+  String get(String name, {String? fallback}) {
+    final value = maybeGet(name, fallback: fallback);
+    assert(() {
```
Comment (ngxingyu) [inline lib/src/dotenv.dart:54]: Does `assert(value != null || fallback != null,'A non-null fallback is required for missing entries');` do the same thing as line 54 to 59?

**Comment by ngxingyu** [comment]
Other than that, the rest of it works fine. perhaps you can also update the Changelog and README and increase the package version in pubspec.yaml

File: lib/src/dotenv.dart (hunk 1 above)
Comment (safarmer) [reply lib/src/dotenv.dart]: It is the same (old habbits). I have made that change and added the readme/changelog updates.

**Comment by safarmer** [comment]
Thanks for the review @ngxingyu . I have made the requested changes.

File: lib/src/dotenv.dart [hunk 2]
```diff
@@ -49,6 +49,14 @@ class DotEnv {
   /// Clear [env]
   void clean() => _envMap.clear();
 
+  String get(String name, {String? fallback}) {
+    final value = maybeGet(name, fallback: fallback);
+    assert(value == null && fallback == null, 'A non-null fallback is required for missing entries');
```
Comment (ngxingyu) [inline lib/src/dotenv.dart:54]: I believe this should be !(value == null && fallback == null) instead

Comment (safarmer) [reply lib/src/dotenv.dart]: Good catch. I have applied a simpler fix since maybeGet will only return null if fallback is null.

**Review by ngxingyu — COMMENTED** [review]
@java-james I've looked through this pr and it seems fine.

**Comment by java-james** [comment]
Thanks @safarmer and thanks @ngxingyu for the review. LGTM too.

**Review by java-james — APPROVED** [review]

**Comment by java-james** [comment]
Thanks for the documentation changes too.

**Pull request merged**